---
title: Home
layout: home
---

| head1        | head two          | three |
|:-------------|:------------------|:------|
| ok           | good swedish fish | nice  |
| out of stock | good and plenty   | nice  |
| ok           | good `oreos`      | hmm   |
| ok           | good `zoute` drop | yumm  |


| Name         | Description/Notability | OS    | License | Site | SBML Support |
|:-------------|:-----------------------|:------|:--------|:-----|:-------------|
| [[iBioSim]]
| iBioSim<ref>{{cite journal |last1=Watanabe |first1=Leandro |last2=Nguyen |first2=Tramy |last3=Zhang |first3=Michael |last4=Zundel |first4=Zach |last5=Zhang |first5=Zhen |last6=Madsen |first6=Curtis |last7=Roehner |first7=Nicholas |last8=Myers |first8=Chris |title=iBioSim3: A Tool for Model-Based Genetic Circuit Design |journal=ACS Synthetic Biology |date=19 July 2019 |volume=8 |issue=7 |pages=1560–1563 |doi=10.1021/acssynbio.8b00078|pmid=29944839 |s2cid=49429947 }}</ref><ref>{{cite journal |last1=Martínez-García |first1=Esteban |last2=Goñi-Moreno |first2=Angel |last3=Bartley |first3=Bryan |last4=McLaughlin |first4=James |last5=Sánchez-Sampedro |first5=Lucas |last6=Pascual del Pozo |first6=Héctor |last7=Prieto Hernández |first7=Clara |last8=Marletta |first8=Ada Serena |last9=De Lucrezia |first9=Davide |last10=Sánchez-Fernández |first10=Guzmán |last11=Fraile |first11=Sofía |last12=de Lorenzo |first12=Víctor |title=SEVA 3.0: an update of the Standard European Vector Architecture for enabling portability of genetic constructs among diverse bacterial hosts |journal=Nucleic Acids Research |date=8 January 2020 |volume=48 |issue=D1 |pages=D1164–D1170 |doi=10.1093/nar/gkz1024|pmid=31740968 |pmc=7018797 }}</ref> is a computer-aided design (CAD) tool for the modeling, analysis, and design of genetic circuits.
||multiplatform (Java/C++)||[[Apache]]||[https://github.com/MyersResearchGroup/iBioSim]|| Yes
|-
| [[CompuCell3D]]
| GUI/Scripting tool<ref>{{cite journal |last1=Swat |first1=Maciej H. |last2=Thomas |first2=Gilberto L. |last3=Belmonte |first3=Julio M. |last4=Shirinifard |first4=Abbas |last5=Hmeljak |first5=Dimitrij |last6=Glazier |first6=James A. |title=Multi-Scale Modeling of Tissues Using CompuCell3D |journal=Methods in Cell Biology |date=2012 |volume=110 |pages=325–366 |doi=10.1016/B978-0-12-388403-9.00013-8|pmid=22482955 |pmc=3612985 |isbn=9780123884039 }}</ref> for building and simulating multicellular models.
||multiplatform (C++/Python)||MIT||[https://compucell3d.org/]|| Yes, but only for reactions.
|-
| [[COPASI]]
| GUI tool<ref>{{cite journal |last1=Bergmann |first1=Frank T. |last2=Hoops |first2=Stefan |last3=Klahn |first3=Brian |last4=Kummer |first4=Ursula |last5=Mendes |first5=Pedro |last6=Pahle |first6=Jürgen |last7=Sahle |first7=Sven |title=COPASI and its applications in biotechnology |journal=Journal of Biotechnology |date=November 2017 |volume=261 |pages=215–220 |doi=10.1016/j.jbiotec.2017.06.1200|pmid=28655634 |pmc=5623632 }}</ref><ref>{{cite journal |last1=Yeoh |first1=Jing Wui |last2=Ng |first2=Kai Boon Ivan |last3=Teh |first3=Ai Ying |last4=Zhang |first4=JingYun |last5=Chee |first5=Wai Kit David |last6=Poh |first6=Chueh Loo |title=An Automated Biomodel Selection System (BMSS) for Gene Circuit Designs |journal=ACS Synthetic Biology |date=19 July 2019 |volume=8 |issue=7 |pages=1484–1497 |doi=10.1021/acssynbio.8b00523|pmid=31035759 |s2cid=140321282 }}</ref> for analyzing and simulating SBML models.
||multiplatform (C++)||Artistic License||[http://www.copasi.org]|| Yes
|-
| Cytosim || Spatial simulator for flexible cytoskeletal filaments and motor proteins<ref>{{cite journal |last1=Nedelec |first1=Francois |last2=Foethke |first2=Dietrich |title=Collective Langevin dynamics of flexible cytoskeletal fibers |journal=New Journal of Physics |volume=9 |date=2007 |issue=11 |pages=427|doi=10.1088/1367-2630/9/11/427 |arxiv=0903.5178 |bibcode=2007NJPh....9..427N |s2cid=16924457 }}</ref> || Mac, Linux, Cygwin (C++) || GPL3 || [http://cytosim.org]|| Not applicable
|-
| [[libroadrunner]]
| High-performance software library for simulation and analysis of SBML models<ref>{{cite journal |last1=Somogyi |first1=Endre T. |last2=Bouteiller |first2=Jean-Marie |last3=Glazier |first3=James A. |last4=König |first4=Matthias |last5=Medley |first5=J. Kyle |last6=Swat |first6=Maciej H. |last7=Sauro |first7=Herbert M. |title=libRoadRunner: a high performance SBML simulation and analysis library: Table 1. |journal=Bioinformatics |date=15 October 2015 |volume=31 |issue=20 |pages=3315–3321 |doi=10.1093/bioinformatics/btv363|pmid=26085503 |pmc=4607739 }}</ref><ref>{{cite journal |last1=Ghaffarizadeh |first1=Ahmadreza |last2=Heiland |first2=Randy |last3=Friedman |first3=Samuel H. |last4=Mumenthaler |first4=Shannon M. |last5=Macklin |first5=Paul |title=PhysiCell: An open source physics-based cell simulator for 3-D multicellular systems |journal=PLOS Computational Biology |date=23 February 2018 |volume=14 |issue=2 |pages=e1005991 |doi=10.1371/journal.pcbi.1005991|pmid=29474446 |pmc=5841829 |bibcode=2018PLSCB..14E5991G }}</ref>
||multiplatform (C/C++)||Apache License||[https://github.com/sys-bio/roadrunner]|| Yes
|-
| [[massPy]]
| Simulation tool <ref>{{cite journal |last1=Haiman |first1=Zachary B. |last2=Zielinski |first2=Daniel C. |last3=Koike |first3=Yuko |last4=Yurkovich |first4=James T. |last5=Palsson |first5=Bernhard O. |title=MASSpy: Building, simulating, and visualizing dynamic biological models in Python using mass action kinetics |journal=PLOS Computational Biology |date=28 January 2021 |volume=17 |issue=1 |pages=e1008208 |doi=10.1371/journal.pcbi.1008208|pmid=33507922 |pmc=7872247 |bibcode=2021PLSCB..17E8208H }}</ref><ref>{{cite journal |last1=Foster |first1=Charles J |last2=Wang |first2=Lin |last3=Dinh |first3=Hoang V |last4=Suthers |first4=Patrick F |last5=Maranas |first5=Costas D |title=Building kinetic models for metabolic engineering |journal=Current Opinion in Biotechnology |date=February 2021 |volume=67 |pages=35–41 |doi=10.1016/j.copbio.2020.11.010|pmid=33360621 |s2cid=229690954 }}</ref>  that can work with COBRApy<ref>{{cite journal |last1=Ebrahim |first1=Ali |last2=Lerman |first2=Joshua A |last3=Palsson |first3=Bernhard O |last4=Hyduke |first4=Daniel R |title=COBRApy: COnstraints-Based Reconstruction and Analysis for Python |journal=BMC Systems Biology |date=December 2013 |volume=7 |issue=1 |pages=74 |doi=10.1186/1752-0509-7-74|pmid=23927696 |pmc=3751080 }}</ref>
||multiplatform (Python)||[[MIT]]||[https://github.com/SBRG/MASSpy]|| Yes
|-
| MCell
| GUI tool for particle-based spatial stochastic simulation with individual molecules<ref>{{cite journal |last1=Stiles |first1=Joel R. |last2=Van Helden |first2=Dirk |last3=Bartol |first3=Thomas M. |last4=Salpeter |first4=Edwin E. |last5=Salpeter |first5=Miriam M |title=Miniature endplate current rise times <100 us from improved dual recordings can be modeled with passive acetylcholine diffusion from a synaptic vesicle |journal=Proc. Natl. Acad. Sci. USA |date=1996 |volume=93 |issue=12 |pages=5747–5752|doi=10.1073/pnas.93.12.5747 |pmid=8650164 |pmc=39132 |doi-access=free }}</ref><ref>{{cite journal |last1=Stiles |first1=Joel R. |last2=Bartol |first2=Thomas M. |title=Monte Carlo methods for simulating realistic synaptic microphysiology using MCell |date=2001 |journal=Computational Neuroscience: Realistic Modeling for Experimentalists |pages=87–127}}</ref><ref>{{cite journal |last1=Kerr |first1=R |last2=Bartol |first2=TM |last3=Kaminsky |first3=B |last4=Dittrich |first4=M |last5=Chang |first5=JCJ |last6=Baden |first6=S |last7=Sejnowski |first7=TJ |last8=Stiles |first8=JR |title=Fast Monte Carlo simulation methods for biological reaction-diffusion systems in solution and on surfaces |date=2008 |journal=SIAM J. Sci. Comput. |volume=30 |issue=6 |pages=3126–3149|doi=10.1137/070692017 |pmid=20151023 |pmc=2819163 |bibcode=2008SJSC...30.3126K }}</ref> || multiplatform || [[MIT]] and [[GPLv2]] || [https://mcell.org/index.html]|| Not applicable
|-
|[https://opencor.ws/ OpenCOR]
|A cross-platform modelling environment, which is aimed at organizing, editing, simulating, and analysing [[CellML]] files on [[Microsoft Windows|Windows]], [[Linux]] and [[macOS]].
|multiplatform (C++/Python)
|[[GPL 3|GPLv3]]
|[https://opencor.ws/]
|Uses [[CellML]]
|-
| [[PhysiBoSS]]
| A specialized form of the PhysiCell agent-based modeling platform that directly integrates Boolean signaling networks into cell Agents<ref>{{cite journal |last1=Letort |first1=Gaelle |last2=Montagud |first2=Arnau |last3=Stoll |first3=Gautier |last4=Heiland |first4=Randy |last5=Barillot |first5=Emmanuel |last6=Macklin |first6=Paul |last7=Zinovyev |first7=Andrei |last8=Calzone |first8=Laurence |title=PhysiBoSS: a multi-scale agent-based modelling framework integrating physical dimension and cell signalling |journal=Bioinformatics |date=1 April 2019 |volume=35 |issue=7 |pages=1188–1196 |doi=10.1093/bioinformatics/bty766|pmid=30169736 |pmc=6449758 }}</ref>
||multiplatform (C++)||[[BSD-3]]||[https://github.com/PhysiBoSS/PhysiBoSS]|| Yes, but only for reactions
|-
| [[PhysiCell]]
| A agent-based<ref>{{cite journal |last1=Ghaffarizadeh |first1=Ahmadreza |last2=Heiland |first2=Randy |last3=Friedman |first3=Samuel H. |last4=Mumenthaler |first4=Shannon M. |last5=Macklin |first5=Paul |title=PhysiCell: An open source physics-based cell simulator for 3-D multicellular systems |journal=PLOS Computational Biology |date=23 February 2018 |volume=14 |issue=2 |pages=e1005991 |doi=10.1371/journal.pcbi.1005991|pmid=29474446 |pmc=5841829 |bibcode=2018PLSCB..14E5991G }}</ref> modeling framework for multicellular systems biology.
||multiplatform (C++)||[[BSD-3]]||[http://physicell.org]|| Yes, but only for reactions
|-
| [[PySCeS]]
| Python tool for modeling and analyzing SBML models<ref>{{cite journal |last1=Olivier |first1=B. G. |last2=Rohwer |first2=J. M. |last3=Hofmeyr |first3=J.-H. S. |title=Modelling cellular systems with PySCeS |journal=Bioinformatics |date=15 February 2005 |volume=21 |issue=4 |pages=560–561 |doi=10.1093/bioinformatics/bti046|pmid=15454409 }}</ref><ref>{{cite journal |last1=Mendoza-Cózatl |first1=David G. |last2=Moreno-Sánchez |first2=Rafael |title=Control of glutathione and phytochelatin synthesis under cadmium stress. Pathway modeling for plants |journal=Journal of Theoretical Biology |date=February 2006 |volume=238 |issue=4 |pages=919–936 |doi=10.1016/j.jtbi.2005.07.003|pmid=16125728 |bibcode=2006JThBi.238..919M }}</ref><ref>{{cite journal |last1=Ghaffarizadeh |first1=Ahmadreza |last2=Heiland |first2=Randy |last3=Friedman |first3=Samuel H. |last4=Mumenthaler |first4=Shannon M. |last5=Macklin |first5=Paul |title=PhysiCell: An open source physics-based cell simulator for 3-D multicellular systems |journal=PLOS Computational Biology |date=23 February 2018 |volume=14 |issue=2 |pages=e1005991 |doi=10.1371/journal.pcbi.1005991|pmid=29474446 |pmc=5841829 |bibcode=2018PLSCB..14E5991G }}</ref>
||multiplatform (Python)||[[BSD-3]]||[https://pysces.sourceforge.net/]|| Yes
|-
| [[pySB]]
| Python-based<ref>{{cite journal |last1=Stefan |first1=Melanie I. |last2=Bartol |first2=Thomas M. |last3=Sejnowski |first3=Terrence J. |last4=Kennedy |first4=Mary B. |title=Multi-state Modeling of Biomolecules |journal=PLOS Computational Biology |date=25 September 2014 |volume=10 |issue=9 |pages=e1003844 |doi=10.1371/journal.pcbi.1003844|pmid=25254957 |pmc=4201162 |bibcode=2014PLSCB..10E3844S }}</ref> platform with specialization in rule-based models.
||multiplatform (Python)||[[BSD-3]]||[https://pysb.org/]|| Partial
|-
| ReaDDy
| Particle-based spatial simulator with intermolecular potentials<ref>{{cite journal |last1=Schöneberg |first1=J. |last2=Ullrich |first2=A. |last3=Noé |first3=F. |date=2014 |title=Simulation tools for particle-based reaction-diffusion dynamics in continuous space |journal=BMC Biophys. |volume=7 |pages=1}}</ref> || Linux and Mac || Custom || [https://readdy.github.io/index.html]|| Not applicable
|-
| [[SBSCL]]
| Java library<ref>{{cite journal |last1=Panchiwala |first1=H |last2=Shah |first2=S |last3=Planatscher |first3=H |last4=Zakharchuk |first4=M |last5=König |first5=M |last6=Dräger |first6=A |title=The Systems Biology Simulation Core Library. |journal=Bioinformatics |date=23 September 2021 |volume=38 |issue=3 |pages=864–865 |doi=10.1093/bioinformatics/btab669 |pmid=34554191|pmc=8756180 }}</ref><ref>{{cite journal |last1=Tangherloni |first1=Andrea |last2=Nobile |first2=Marco S. |last3=Cazzaniga |first3=Paolo |last4=Capitoli |first4=Giulia |last5=Spolaor |first5=Simone |last6=Rundo |first6=Leonardo |last7=Mauri |first7=Giancarlo |last8=Besozzi |first8=Daniela |title=FiCoS: A fine-grained and coarse-grained GPU-powered deterministic simulator for biochemical networks |journal=PLOS Computational Biology |date=9 September 2021 |volume=17 |issue=9 |pages=e1009410 |doi=10.1371/journal.pcbi.1009410|pmid=34499658 |pmc=8476010 |bibcode=2021PLSCB..17E9410T }}</ref> with efficient and exhaustive support for SBML
||multiplatform (Java)||[[LGPL]]||[https://draeger-lab.github.io/SBSCL/]|| Yes
|-
| [[SBW (software)|SBW]]
| A distributed workbench<ref>{{cite journal |last1=Hucka |first1=M. |last2=Finney |first2=A. |last3=Sauro |first3=H. M. |last4=Bolouri |first4=H. |last5=Doyle |first5=J. |last6=Kitano |first6=H. |title=The Erato Systems Biology Workbench: Enabling Interaction and Exchange Between Software Tools for Computational Biology |journal=Biocomputing 2002 |date=December 2001 |pages=450–461 |doi=10.1142/9789812799623_0042|pmid=11928498 |isbn=978-981-02-4777-5 }}</ref><ref>{{cite journal |last1=Kawasaki |first1=Regiane |last2=Baraúna |first2=Rafael A. |last3=Silva |first3=Artur |last4=Carepo |first4=Marta S. P. |last5=Oliveira |first5=Rui |last6=Marques |first6=Rodolfo |last7=Ramos |first7=Rommel T. J. |last8=Schneider |first8=Maria P. C. |title=Reconstruction of the Fatty Acid Biosynthetic Pathway of Exiguobacterium antarcticum B7 Based on Genomic and Bibliomic Data |journal=BioMed Research International |date=2016 |volume=2016 |pages=1–9 |doi=10.1155/2016/7863706|pmid=27595107 |pmc=4993939 |doi-access=free }}</ref> that includes many modeling tools
||multiplatform (C/C++)||[[BSD-3]]||[https://sbw.sourceforge.net/]|| Yes
|-
| Smoldyn
| Particle-based simulator for spatial stochastic simulations with individual molecules<ref>{{cite journal |last1=Andrews |first1=Steven S. |last2=Bray |first2=Dennis |title=Stochastic simulation of chemical reactions with spatial resolution and single molecule detail |journal=Physical Biology |date=2004 |volume=1 |issue=3–4 |pages=137–151|doi=10.1088/1478-3967/1/3/001 |pmid=16204833 |bibcode=2004PhBio...1..137A |s2cid=16394428 }}</ref><ref>{{cite journal |last1=Andrews |first1=Steven S. |last2=Addy |first2= Nathan J. |last3= Brent |first3=Roger |last4=Arkin |first4=Adam P. |title=Detailed simulations of cell biology with Smoldyn 2.1 |journal=PLOS Comp. Biol. |date=2010 |volume=6 |issue=3 |pages=e1000705|doi=10.1371/journal.pcbi.1000705 |pmid=20300644 |pmc=2837389 |bibcode=2010PLSCB...6E0705A }}</ref><ref>{{cite journal |last1=Andrews |first1=Steven S. |title=Smoldyn: particle-based simulation with rule-based modeling, improved molecular interaction, and a library interface |journal=Bioinformatics |date=2017 |volume=33 |issue=5 |pages=710–717|doi=10.1093/bioinformatics/btw700 |pmid=28365760 }}</ref><ref>{{cite journal |last1=Singh |first1 =Dilawar |last2=Andrews |first2=Steven S. |title=Python interfaces for the Smoldyn simulator |journal=Bioinformatics |date=2022 |volume=38 |pages=291–293|doi =10.1093/bioinformatics/btab530 |pmid =34293100 }}</ref>
|| multiplatform (C/C++/Python)||[[LGPL]]||[https://www.smoldyn.org/]||Not applicable
|-
| Spatiocyte
| Spatial modeling software that uses a fine lattice with up to one molecule per site<ref>{{cite book |last1=Arjunan |first1=S.N.V. |last2= Takahashi |first2= K. |date=2017 |title=Multi-algorithm particle simulations with Spatiocyte |series=Methods in Molecular Biology |volume=1611 |pages=219–236}}</ref><ref>{{cite journal |last1=Arjunan |first1=S.N.V. |last2= Miyauchi |first2=A. |last3=Iwamoto |first3= K. |last4=Takahashi |first4=K. |date=2020 |title=pSpatiocyte: a high-performance simulator for intracellular reaction-diffusion systems |journal=BMC Bioinformatics |volume=21 |issue=1 |pages=33|doi=10.1186/s12859-019-3338-8 |pmid=31996129 |pmc=6990473 }}</ref> || multiplatform || Unknown || [https://spatiocyte.org]|| Not applicable
|-
| SpringSaLaD
| Particle-based spatial simulator in which molecules are spheres that are linked by springs<ref>{{cite journal |last1=Michalski |first1=P.J. |last2=Loew |first2=L.M. |date=2016 |title=SpringSaLaD: a spatial, particle-based biochemical simulation platform with excluded volume |journal=Biophys. J. |volume=110 |issue=3 |pages=523–529|doi=10.1016/j.bpj.2015.12.026 |pmid=26840718 |pmc=4744174 |bibcode=2016BpJ...110..523M }}</ref> || multiplatform || Unknown || [https://vcell.org/ssalad]|| Not applicable
|-
|STEPS
|Stochastic reaction-diffusion and membrane potential solver on distributed meshes<ref>{{cite journal |last1=Hepburn |first1=Iain |last2=Chen |first2=Weiliang |last3=Wils |first3=Stefan |last4=De Schutter |first4=Erik |title=STEPS: efficient simulation of stochastic reaction–diffusion models in realistic morphologies |journal=BMC Systems Biology |date=May 2012 |volume=7 |issue=1 |pages=36 |doi=10.1186/1752-0509-6-36|pmid=22574658 |s2cid=9165862 }}</ref><ref>{{cite journal |last1=Chen |first1=Weiliang |last2=De Schutter |first2=Erik |title=Parallel STEPS: Large Scale Stochastic Spatial Reaction-Diffusion Simulation with High Performance Computers |journal=Frontiers in Neuroinformatics |date=February 2017 |volume=11 |issue=1 |pages=13 |doi=10.3389/fninf.2017.00013|pmid=28239346 |pmc=5301017 |doi-access=free }}</ref><ref>{{cite journal |last1=Hepburn |first1=Iain |last2=Chen |first2=Weiliang |last3=De Schutter |first3=Erik |title=Accurate reaction-diffusion operator splitting on tetrahedral meshes for parallel stochastic molecular simulations |journal=The Journal of Chemical Physics |date=August 2016 |volume=145 |issue=5 |pages=054118 |doi=10.1063/1.4960034|pmid=27497550 |arxiv=1512.03126 |bibcode=2016JChPh.145e4118H |s2cid=17356298 }}</ref><ref>{{cite journal |last1=Chen |first1=Weiliang |last2=Carel |first2=Tristan |last3=Awile |first3=Omar |last4=Cantarutti |first4=Nicola |last5=Castiglioni |first5=Giacomo |last6=Cattabiani |first6=Alessandro |last7=Del Marmol |first7=Baudouin |last8=Hepburn |first8=Iain |last9=King |first9=James G. |last10=Kotsalos |first10=Christos |last11=Kumbhar |first11=Pramod |last12=Lallouette |first12=Jules |last13=Melchior |first13=Samuel |last14=Schürmann |first14=Felix |last15=De Schutter |first15=Erik |title=STEPS 4.0: Fast and memory-efficient molecular simulations of neurons at the nanoscale |journal=Frontiers in Neuroinformatics |date=October 2022 |volume=16  |doi=10.3389/fninf.2022.883742 |issn=1662-5196|doi-access=free }}</ref>
|multiplatform (C++/Python)
|[[GPLv2]]
|[https://steps.sourceforge.net]
|Partial [https://steps.sourceforge.net/manual/sbml_importer.html#Level-of-support]
|-
| [[Tellurium (software)|Tellurium]]
| Simulation environment,<ref>{{cite journal |last1=Choi |first1=Kiri |last2=Medley |first2=J. Kyle |last3=König |first3=Matthias |last4=Stocking |first4=Kaylene |last5=Smith |first5=Lucian |last6=Gu |first6=Stanley |last7=Sauro |first7=Herbert M. |title=Tellurium: An extensible python-based modeling environment for systems and synthetic biology |journal=Biosystems |date=September 2018 |volume=171 |pages=74–79 |doi=10.1016/j.biosystems.2018.07.006|pmid=30053414 |pmc=6108935 }}</ref><ref>{{cite journal |last1=Pease |first1=Nicholas A. |last2=Nguyen |first2=Phuc H.B. |last3=Woodworth |first3=Marcus A. |last4=Ng |first4=Kenneth K.H. |last5=Irwin |first5=Blythe |last6=Vaughan |first6=Joshua C. |last7=Kueh |first7=Hao Yuan |title=Tunable, division-independent control of gene activation timing by a polycomb switch |journal=Cell Reports |date=March 2021 |volume=34 |issue=12 |pages=108888 |doi=10.1016/j.celrep.2021.108888|pmid=33761349 |pmc=8024876 }}</ref> that packages multiple libraries into one platform.
||multiplatform (Python)||Apache License||[https://github.com/sys-bio/tellurium]|| Yes
|-
| URDME
| Stochastic reaction-diffusion simulation on unstructured meshes<ref>{{cite journal |last1=Drawert |first1=B. |last2=Engblom |first2=S. |last3=Hellander |first3=A |title=URDME: A modular framework for stochastic simulation of reaction-transport processes in complex geometries |journal=BMC Systems Biology |volume=6 |date=2012|page=76 |doi=10.1186/1752-0509-6-76 |pmid=22727185 |pmc=3439286 }}</ref> || MatLab on Mac, Linux || GPL3 || [http://urdme.github.io/urdme/]|| Not applicable
|-
| [[VCell]]
| Comprehensive modeling platform<ref>{{cite journal |last1=Schaff |first1=J. |last2=Fink |first2=C.C. |last3=Slepchenko |first3=B. |last4=Carson |first4=J.H. |last5=Loew |first5=L.M. |title=A general computational framework for modeling cellular structure and function |journal=Biophysical Journal |date=September 1997 |volume=73 |issue=3 |pages=1135–1146 |doi=10.1016/S0006-3495(97)78146-3|bibcode=1997BpJ....73.1135S |s2cid=39818739 }}</ref><ref>{{cite journal |last1=Cowan |first1=Ann E. |last2=Moraru |first2=Ion I. |last3=Schaff |first3=James C. |last4=Slepchenko |first4=Boris M. |last5=Loew |first5=Leslie M. |title=Spatial Modeling of Cell Signaling Networks |journal=Methods in Cell Biology |date=2012 |volume=110 |pages=195–221 |doi=10.1016/B978-0-12-388403-9.00008-4|pmid=22482950 |pmc=3519356 |isbn=9780123884039 }}</ref> for non-spatial, spatial, deterministic and stochastic simulations, including both reaction networks and reaction rules. 
||multiplatform (Java)||MIT||[https://vcell.org]|| Yes
|-
|
|



This is a *bare-minimum* template to create a Jekyll site that uses the [Just the Docs] theme. You can easily set the created site to be published on [GitHub Pages] – the [README] file explains how to do that, along with other details.

If [Jekyll] is installed on your computer, you can also build and preview the created site *locally*. This lets you test changes before committing them, and avoids waiting for GitHub Pages.[^1] And you will be able to deploy your local build to a different platform than GitHub Pages.

More specifically, the created site:

- uses a gem-based approach, i.e. uses a `Gemfile` and loads the `just-the-docs` gem
- uses the [GitHub Pages / Actions workflow] to build and publish the site on GitHub Pages

Other than that, you're free to customize sites that you create with this template, however you like. You can easily change the versions of `just-the-docs` and Jekyll it uses, as well as adding further plugins.

[Browse our documentation][Just the Docs] to learn more about how to use this theme.

To get started with creating a site, just click "[use this template]"!

----

[^1]: [It can take up to 10 minutes for changes to your site to publish after you push the changes to GitHub](https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll#creating-your-site).

[Just the Docs]: https://just-the-docs.github.io/just-the-docs/
[GitHub Pages]: https://docs.github.com/en/pages
[README]: https://github.com/just-the-docs/just-the-docs-template/blob/main/README.md
[Jekyll]: https://jekyllrb.com
[GitHub Pages / Actions workflow]: https://github.blog/changelog/2022-07-27-github-pages-custom-github-actions-workflows-beta/
[use this template]: https://github.com/just-the-docs/just-the-docs-template/generate
